### DISCUSSION TITLE
Syntax and structure of osm.txt

### DISCUSSION THREAD

**Discussion by lenz-exe** [karttapullautin/karttapullautin#117] [Q&A] [answered]
Can someone explain the syntax and structure of the `osm.txt `file to me, as I somehow can't quite figure it out

Thank you so much...

**Comment by rphlo** [comment, marked as answer]
the vectorconf files are "pipe" ("|") separated files consisting of 3 columns.
The first column is a comment of what element the line describe.
The second column is the ISOM2000 code of the element (ISOM is the orienteering standard for maps) you can find the specs here 
[isom-2000.pdf](https://github.com/user-attachments/files/16812582/isom-2000.pdf)

The 3rd column is the logic that should be applied to determine if a shape file feature match that element

Examples from the osm.txt:
"power line|524|barrier!=" states that we will draw the isom code 524 if the tag of the shapefile feature contains a key named "barrier" that has a value different than a empty string

"playground|529|leisure=playground" states we will draw a area with the isom code 529 if the shapefile feature contains a tag with key "leisure" that the value is equal to "playground"

"road-path|503|highway=motorway&bridge!=yes" states we will draw a area with the isom code 503 if the shapefile feature contains a tag with key "highway" that the value is equal to "motorway" and does contains a key "bridge" with a value not being "yes"

**Comment by lenz-exe** [comment]
Thank you for your prompt response.

I noticed a potential discrepancy between the ISOM codes in the Rust code and the provided PDF file. Specifically, in the Rust code, there's this line [link](https://github.com/rphlo/karttapullautin/blob/ed5c506e56d32f46ef28feb94f28a1385b867ade/src/main.rs#L1869C1-L1870C43):
```rust
// buildings
if isom == "526" {
```
However, according to the PDF, ISOM code 526 corresponds to a "Cairn (P)" which is defined as "A prominent cairn, memorial stone, boundary stone, or trigonometric point."

**Reply by rphlo** [reply]
Sorry, actually the reference codes for ISOM are from the previous version known as ISOM 2000 See: http://lazarus.elte.hu/mc/specs/isom-2000.pdf

**Comment by rphlo** [comment]
[isom-2000.pdf](https://github.com/user-attachments/files/16812572/isom-2000.pdf)
Attaching the ISOM spec mentioned in this thread in case the link goes missing.
